### PR TITLE
Add disabled state to tabs

### PR DESCRIPTION
**Pull request by soupette**
Signed-off-by: soupette <cyril@strapi.io>

# [title]

## Description
[add description]

## Demo
Example on : [deployed story link]

## API
[api details]

```jsx
<YourComponent />
```

## Voiceover

[voiceover gif]

### CODE REVIEW

File: packages/strapi-parts/src/Tabs/Tabs.js [hunk 1]
```diff
@@ -97,25 +108,28 @@ export const Tab = ({ selected, id, children, variant, hasError, onClick, onTabC
...
-      <button
+      <TabButton
         id={tabId}
         role="tab"
         aria-controls={selected ? tabPanelId : undefined}
         tabIndex={selected ? 0 : -1}
         aria-selected={selected}
         type="button"
         onClick={handleClick}
+        aria-disabled={disabled}
```
Comment (HichamELBSI) [inline packages/strapi-parts/src/Tabs/Tabs.js:126]: Don't we need to pass the disabled props also ?

```js
disabled={disabled}
```

File: packages/strapi-parts/src/Tabs/Tabs.js [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
 import { ButtonText, TableLabel } from '../Text';
 import { KeyboardKeys } from '../helpers/keyboardKeys';
 import { useTabsFocus } from './useTabsFocus';
 import { DefaultTabsRow, DefaultTabButton, DefaultTabBox, SimpleTabBox } from './components';
 
+const TabButton = styled.button`
+  &[aria-disabled='true'] {
+    cursor: not-allowed;
+  }
+`;
```
Comment (HichamELBSI) [inline packages/strapi-parts/src/Tabs/Tabs.js:14]: Why do we need this modification ?

Comment (HichamELBSI) [reply packages/strapi-parts/src/Tabs/Tabs.js:14]: If we do this here, don't we need to standardize it in every disabled components ? @maevalienard

File: packages/strapi-parts/src/Tabs/Tabs.js (hunk 1 above)
Comment (mfrachet) [reply packages/strapi-parts/src/Tabs/Tabs.js:126]: disabled=disabled is not practical for people using keyboard. When using disabled, the field is not tabbable and so not announced to the users. On the other hand, relying on aria-disabled and preventing the real click allows for the button to be tabbable and announced as "dimmed" by SRs

**Pull request merged**